File: ddd/tests/fixtures/operation-error-set/typescript-class-workspace/billing-domain/src/undeclared.ts
/** The mapped type exists, but neither mapped operation does. */

import type { Result } from "./result.ts";

export type IssueInvoiceError = "already-issued" | "empty-lines";

export class Invoice {
  static restore(): Invoice {
    return new Invoice();
  }

  close(): Result<void, IssueInvoiceError> {
    return { ok: true, value: undefined };
  }
}

File: ddd/tests/fixtures/operation-error-set/rust-workspace/billing-domain/src/undeclared.rs
//! The mapped type exists, but neither mapped operation does.

pub enum IssueInvoiceError {
    AlreadyIssued,
    EmptyLines,
}

pub struct Invoice;

impl Invoice {
    pub fn close(&mut self) -> Result<(), IssueInvoiceError> {
        Ok(())
    }
}
